File: XamarinStore.Droid/Views/CircleDrawable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Graphics;
using Android.Graphics.Drawables;
using Android.Animation;

namespace XamarinStore
{
	public class CircleDrawable : Drawable
	{
		Bitmap bmp;
		BitmapShader bmpShader;
		Paint paint;
		RectF oval;

		public CircleDrawable (Bitmap bmp)
		{
			this.bmp = bmp;
			this.bmpShader = new BitmapShader (bmp, Shader.TileMode.Clamp, Shader.TileMode.Clamp);
			this.paint = new Paint () { AntiAlias = true };
			this.paint.SetShader (bmpShader);
			this.oval = new RectF ();
		}

		public override void Draw (Canvas canvas)
		{
			canvas.DrawOval (oval, paint);
		}

		protected override void OnBoundsChange (Rect bounds)
		{
			base.OnBoundsChange (bounds);
			oval.Set (0, 0, bounds.Width (), bounds.Height ());
		}

		public override int IntrinsicWidth {
			get {
				return bmp.Width;
			}
		}

		public override int IntrinsicHeight {
			get {
				return bmp.Height;
			}
		}

		public override void SetAlpha (int alpha)
		{

		}

		public override int Opacity {
			get {
				return (int)Format.Opaque;
			}
		}

		public override void SetColorFilter (ColorFilter cf)
		{

		}
	}
}

File: F#/XamarinStore.Droid/Drawables/CircleDrawable.fs
namespace XamarinStore

open Android.Graphics
open Android.Graphics.Drawables

type CircleDrawable(bmp:Bitmap) =
    inherit Drawable()

    let bmpShader = new BitmapShader (bmp, Shader.TileMode.Clamp, Shader.TileMode.Clamp)
    let paint = new Paint (AntiAlias = true )
    do paint.SetShader bmpShader |> ignore
    let oval = new RectF()

    override this.Draw canvas =
        canvas.DrawOval (oval, paint)

    override this.OnBoundsChange bounds =
        base.OnBoundsChange bounds
        oval.Set(0.0f, 0.0f, float32 (bounds.Width()), float32 (bounds.Height()))

    override this.IntrinsicWidth = bmp.Width

    override this.IntrinsicHeight = bmp.Height

    override this.SetAlpha alpha =
        ()

    override this.Opacity = int Format.Opaque

    override this.SetColorFilter cf =
        ()


